### ISSUE TITLE
Quotes on shortcode parameters are being escaped

### ISSUE THREAD

**Issue by dartiss** [WordPress/gutenberg#4692]
I'm using WordPress 4.9.2 and Gutenberg 2.1.0.

I'm finding when I'm trying to add shortcode parameters that need quotes (double, single, or whatever) around them, that they're being escaped.

So, I use a plugin that allows me, via a shortcode, to add updates to a post. If I add the following into a block...

`[update title="The latest update"]Some text here[/update]`

This is showing correctly visually but when switching to HTML is shows as...

`[update title=&quot;The latest update&quot;]Some text here[/update]`

Obviously, when viewed on the site, it displays incorrectly because of the escaping.

Now, this doesn't happen if the shortcode is added as a shortcode block - only if it's added inline or written out manually into a paragraph, for example.

**Comment by alanef** [comment]
WP 4.9.4  and 2.2.0

Gutenberg passes escape quotes to the shortode atts - the consquence is if the atts are used in JS the JS breaks

Switch to classic = no issue

**Comment by bobbingwide** [comment]
I'm trying to write a block that will generate the shortcode so that I can re-use an existing solution in a hybrid environment where the content was created in the classic-editor. 
The HTML for the shortcode is generated correctly by the edit method ... it's displayed correctly in edit. 
But on saving double quotes gets converted to &quot; and single quotes get converted to &#x27;

**Comment by mattyrob** [comment]
@bobbingwide 

I might be able to help with this - I have a block for a shortcode that I'm trying to give a UI to for the parameters. The deprecation of allowing HTML to be passed and forcing RawHTML or createElement is causing me problems. Ping me if you want my code.

**Comment by bobbingwide** [comment]
@mattyrob  Perhaps you'd like to compare your code with mine.  
In my Countdown block I did manage to implement a solution using `<RawHTML>` 
See https://github.com/bobbingwide/oik-block/blob/master/blocks/oik-countdown/index.js

**Comment by mattyrob** [comment]
@bobbingwide 

Thanks for that link - my code is in ES5 so I'm trying to call RawHTML as a function and it's not really working at the moment. I don't think `<RawHTML>` will work in my code.

**Comment by rhyswynne** [comment]
I'm having the same issue as @mattyrob. My blocks right now are outputting a Title, introduction paragraph then a shortcode. My save function is this:-

`return ( el( 'div', { className: props.className },
					el( 'h2', { className: 'wpec-title'}, attributes.title ),
					el( 'p', { className: 'wpec-description' }, attributes.subscriptiondetails ),
					wp.element.RawHTML( { props , emailcaptureform } )
					)
					);
				},
			} );`

The problem arises is when output instead of the form, we're getting the following:-

`<div props="[object Object]" emailcaptureform="[email_capture_form  buttontext=&quot;Test%20Button&quot;]"></div>`

I don't think I need `props` (In fact I'm fairly confident I don't), but it would be good to get an example of how wp.element.RawHTML will work, as most code is ES6.

**Comment by mattyrob** [comment]
@rhyswynne

I ended up using the following:
`return el( element.RawHTML, null, '<p>my unescaped text here</p>' );`

I hope that helps you.

**Comment by rhyswynne** [comment]
Thank you! It did :)

**Comment by bobbingwide** [comment]
At WordCamp London contributor day @dartiss confirmed that #5897 resolves the issues with quotes in shortcodes being serialized. It resolve the issue for both double and single quotes.
The PR does not resolve issue #4456

**Comment by mcsf** [comment]
Closed via #5897

**Comment by aligoren** [comment]
I faced the same issue. What should I do? Should I use the classic editor?

If I use the classic editor why I'm using Gutenberg now and why the Gutenberg coming as default editor?

This isn't a feature. This is a bug.

Now, I'll install classic editor because of the damn escape sequence.

I Tried This Code with Markdown

```js
I.executeScript(() => {
	const el = document.querySelector('#tarih1')

	const event = new Event('change');

	el.value = '2018-12-01'

	el.dispatchEvent(event)
})
```

And it's output like below;

```html
I.executeScript(() =&gt; { 
   const el = document.querySelector('#tarih1')  
   const event = new Event('change');  
   el.value = '2018-12-01'  el.dispatchEvent(event)
})

```

What should I do?

There is another problem when I'm using Firefox.

My browser saying "Dad! I'm dying.". The Gutenberg works so slow after 3-4 paragraphs.
